src/main/kotlin/app/revanced/patches/tiktok/misc/simspoof/patch/SimSpoofPatch.kt

```diff
@@ -0,0 +1,103 @@
+import app.revanced.patches.tiktok.misc.settings.patch.TikTokSettingsPatch
+import app.revanced.patches.tiktok.misc.simspoof.annotations.SimSpoofCompatibility
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.ClassDef
+import org.jf.dexlib2.iface.Method
+import org.jf.dexlib2.iface.instruction.OneRegisterInstruction
+import org.jf.dexlib2.iface.instruction.formats.Instruction35c
+import org.jf.dexlib2.iface.reference.MethodReference
+import org.jf.dexlib2.immutable.reference.ImmutableMethodReference
+
+@Patch(false)
+@DependsOn([TikTokIntegrationsPatch::class, TikTokSettingsPatch::class])
+@Name("sim-spoof")
+@Description("Fakes the sim card information to bypass region restrictions.")
+@SimSpoofCompatibility
+@Version("0.0.1")
+class SimSpoofPatch : BytecodePatch() {
+
+    internal companion object {
+        private val PATCH_METHOD = hashMapOf(
+            "getSimCountryIso" to "getCountryIso",
+            "getNetworkCountryIso" to "getCountryIso",
+            "getSimOperator" to "getOperator",
+            "getNetworkOperator" to "getOperator",
+            "getSimOperatorName" to "getOperatorName",
+            "getNetworkOperatorName" to "getOperatorName"
+        )
+
+        private fun MutableClass.findMutableMethodOf(
+            method: Method
+        ) = this.methods.first {
+            it.softCompareTo(
+                ImmutableMethodReference(
+                    method.definingClass, method.name, method.parameters, method.returnType
+                )
+            )
+        }
+    }
+
+    override fun execute(context: BytecodeContext): PatchResult {
+        //Find all api call to check sim information
+        context.classes.forEach { classDef ->
+            classDef.methods.forEach br@{ method ->
+                with(method.implementation) {
+                    this?.instructions?.forEach cont@{
+                        if (it.opcode != Opcode.INVOKE_VIRTUAL) return@cont
+                        val methodRef = (it as Instruction35c).reference as MethodReference
+                        if (methodRef.definingClass != "Landroid/telephony/TelephonyManager;") return@cont
+                        if (!PATCH_METHOD.containsKey(methodRef.name)) return@cont
+                        patchAPI(context, classDef, method)
+                        return@br
+                    }
+                }
+            }
+        }
+        //Enable sim-spoof in settings
+        with(SettingsStatusLoadFingerprint.result!!.mutableMethod) {
+            addInstruction(
+                0,
+                "invoke-static {}, Lapp/revanced/tiktok/settingsmenu/SettingsStatus;->enableSimSpoof()V"
+            )
+        }
+        return PatchResultSuccess()
+    }
+
+    //Patch android API and return fake sim information
+    private fun patchAPI(context: BytecodeContext, classDef: ClassDef, method: Method) {
+        with(context.proxy(classDef).mutableClass.findMutableMethodOf(method)) {
```

Comment:
Instead of having to iterate over all instructions again, you can just pass the index where the instruction you want to patch is located at.

Reply:
I did as you suggested before, but I ran into the issue that when two places in a method need to be patched, if the first place is patched, the index of the second place will change. Can you suggest a way to solve this problem?

Reply:
Collect all indices and pass them as an array. This method will add instructions in reverse order, so from bottom to top.